### PR TITLE
Improved main.py and created configuration.json 

### PR DESCRIPTION
**Pull request by muthuubalakan**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -8,69 +8,98 @@
...
     This function will check for valid cookie on all pages!
     Return True if there is valid cookie.
-
     """
     cookie = request.cookies.get(COOKIE_NAME, None)
-    if not cookie:
-        return False
-    if COOKIE_VALUE == 0:
+    # If there is a cookie, return True.
+    if cookie:
```
Comment (hgn) [inline main.py]: no. what do you want to check? That there *IS* a cookie? No you want to check the cookie value!

File: main.py [hunk 2]
```diff
@@ -8,69 +8,98 @@
...
 
 
+# Check for the requested file in the system, return server error! if file not found.
+def check_file(filename):
+    if not os.path.isfile(filename):
+        print("Internal server error! file not found: {}".format(filename))
+        return False
+    response = open(filename).read()
+    return response
+
```
Comment (lwrl-zz) [inline main.py]: naming is unclear here: check_file should check the file, not more. name your functions after their purpose!

File: main.py [hunk 3]
```diff
@@ -8,69 +8,98 @@
...
+
+# User credentials stored in a json file. Loads the json file to access the values.
+# Check if json file can be accessible.
+check = check_file('configuration.json')
+if check:
+    users = open('configuration.json')
```
Comment (lwrl-zz) [inline main.py]: does the json file really only contain users? if not, adjust return value parameter!

File: main.py [hunk 4]
```diff
@@ -8,69 +8,98 @@
...
+    if not os.path.isfile(filename):
+        print("Internal server error! file not found: {}".format(filename))
+        return False
+    response = open(filename).read()
+    return response
+
+
+# User credentials stored in a json file. Loads the json file to access the values.
+# Check if json file can be accessible.
+check = check_file('configuration.json')
```
Comment (lwrl-zz) [inline main.py]: majbe also user define for configuration file name?!

File: main.py [hunk 5]
```diff
@@ -8,69 +8,98 @@
...
+    # If there is a cookie, return True.
+    if cookie:
         return True
+    return False
 
 
+# Get the user credentials from json data and validate the user inputs.
 def check_credentials(username, password):
-    if username == USERNAME and password == PASSWORD:
+    if username == load_data['USERNAME'] and password == load_data['PASSWORD']:
```
Comment (lwrl-zz) [inline main.py]: check if a param exist before accessing values! can you be sure that load_data really exists here?!

File: main.py [hunk 6]
```diff
@@ -1,112 +1,182 @@
...
+# Cookie safety has been implemented. If user tries to
+# modify the cookie value it returns False.
+# Make sure user cannot modify cookie expire date.
 def check_cookie(request):
-    """cookie_check!
-    This function will check for valid cookie on all pages!
-    Return True if there is valid cookie.
-
-    """
     cookie = request.cookies.get(COOKIE_NAME, None)
```
Comment (lwrl-zz) [inline main.py]: you get the cookie value here, not the cookie. naming should be clear!

File: main.py [hunk 7]
```diff
@@ -1,112 +1,182 @@
...
     if not cookie:
         return False
-    if COOKIE_VALUE == 0:
+    try:
+        if not int(cookie) == COOKIE_VALUE:
+            return False
         return True
+    except ValueError:
```
Comment (lwrl-zz) [inline main.py]: we might not need this here. anyway, its not working. what is the intention here? do you want an error log for the developer? think we can remove this...

**Pull request merged**